.github/workflows/go.yml

```
@@ -0,0 +1,86 @@
+name: Go
+
+on: [push]
```

This is fired only when something is pushed to this repository. Pull Requests from other repository needs `pull_request` trigger.
https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request
```suggestion
on:
  pull_request: {}
  push:
    branches: [master]
```